src/huggingface_hub/hub_mixin.py

```diff
@@ -145,7 +124,6 @@ def from_pretrained(
         """
 
         model_id = pretrained_model_name_or_path
-        map_location = torch.device(map_location)
```

Comment:
Unrelated to your PR, but some good cleanup I think: `hf_hub_url` + `cached_download` can be simplified with `hf_hub_download`. Feel free to do it in a separate PR if you prefer. 

(this is few lines below)

Reply:
Yes, I'll have to give that a go. I'll make separate PR :smile: